include/xtensor/xeval.hpp

```diff
@@ -47,6 +47,65 @@ namespace xt
     {
         return std::forward<T>(t);
     }
+
+    /**************
+     * as_strided *
+     **************/
+
+    /**
+     * Force evaluation of xexpression and convert to the required layout.
+     * @return xarray or xtensor depending on shape type
```

Comment:
This is not exact, if the input parameter is a view with the specified layout, `as_strided` does not evaluate anything, it simply forwards the view. This makes me wonder if we should add this function that returns a view (not holding the data, with possible dangling reference issue) or a container depending on its argument.

I understand the use case (you pay not want to evaluate a view that van be directly used if it already provides the data interface) but I really wonder if it is worth adressing it.

Reply:
> This makes me wonder if we should add this function that returns a view (not holding the data, with possible dangling reference issue) or a container depending on its argument.

In that case, that would be equivalent to only extend `xt::eval` with the capability to change the layout if I'm right.
So, do we need it ?

Reply:
yes, we actually need it, but we should add a warning in the documentation saying that this function should be used in a local context only (i.e. never return the value returned by this function).

Reply:
great thanks